### PR TITLE
[DEV-1166] Endpoint to retrieve all vaults

### PR DESCRIPTION
**Pull request by Krysh90**

### CODE REVIEW

**Review by SwissBlueDragon — APPROVED** [review]

File: src/subdomains/yield-machine/application/api/controllers/yield-machine.controller.ts [hunk 1]
```diff
@@ -18,4 +18,10 @@ export class YieldMachineController {
   createTransaction(@Body() dto: TransactionInputDto): Promise<string> {
     return this.yieldMachineService.create(dto);
   }
+
+  @Get('vaults')
+  @ApiExcludeEndpoint()
+  getVaults(): Promise<string[]> {
+    return this.yieldMachineService.getVaults();
+  }
```
Comment (davidleomay) [inline src/subdomains/yield-machine/application/api/controllers/yield-machine.controller.ts:26]: Wouldn't it make sense to move this to the vault subdomain `src/integration/vault` (same as for masternodes)?

Comment (Krysh90) [reply src/subdomains/yield-machine/application/api/controllers/yield-machine.controller.ts:26]: thought that those are yield-machine vaults and therefore it makes sense to access them via yield-machine endpoint

Comment (Krysh90) [reply src/subdomains/yield-machine/application/api/controllers/yield-machine.controller.ts:26]: the idea was to show at least the count of LOCK vaults on vault-maxi statistics page, and we might want to show more stats in comparison to community projects. If so we would need the yield machine vaults.

Comment (davidleomay) [reply src/subdomains/yield-machine/application/api/controllers/yield-machine.controller.ts:26]: But then we would want all vaults of LOCK?

Comment (Krysh90) [reply src/subdomains/yield-machine/application/api/controllers/yield-machine.controller.ts:26]: are there others than yield-machine vaults? will there be? I don't know 🤷‍♂️

**Review by davidleomay — APPROVED** [review]

**Pull request merged**